### ISSUE TITLE
Hi, about the experiment 

### ISSUE THREAD

**Issue by ross-Hr** [ZhichengHuang/CMAE#4]
Hi,
```
All pre-training experiments are conducted on 32 NVIDIA A100 GPUs with a batch size of 4096.
```
If i set batch_sizes=16 , is it a devastating blow to the experimental results ?

**Comment by HuangChiEn** [comment]
> Hi,
> 
> ```
> All pre-training experiments are conducted on 32 NVIDIA A100 GPUs with a batch size of 4096.
> ```
> 
> If i set batch_sizes=16 , is it a devastating blow to the experimental results ?

if you don't mind the non-official reply.
Basically, the smaller batch_size often harmful the pertaining procedure of SSL (Self-Supervised Learning) methods, even some proposed method is robust to smaller one in some degree.

For SSL based method, 4096 batch-size is regarded as an golden standard, the smaller batch size (s.t. 2048, 1024) is still acceptable and may stabilize (2048 batch-size) the pertaining procedure in my experience (although, lower downstream accuracy is given).  
 
However, any batch-sizes lower then 1024 is out of the suitable range for SSL pertaining. In the case you have mentioned batch-size=16 may work terrible i think.

---------------------------------------------------------------------------------

Solution : 
If your code is implemented by pytorch-lightning, the parameters `accumulated_grad` of `pl.Trainer` maybe suitable for your case.
As the name implies, `accumulated_grad` conduct the backpropogation with accumulating the enough batch-size. 
In your case, suppose batch-size=16, and you want to simulate 2048 (16 x 128=2048) batch-size pretraining, you can set `pl.Trainer(..., accumulated_grad=128)`. Then all will going fine (except it may take more time to perform the forward step of model).

**Comment by ross-Hr** [comment]
Great, thanks!
I will try this solution. 
@HuangChiEn

**Comment by AlexNmSED** [comment]
@ross-Hr Hi, I am trying to implement this work, but I am encountering many problems. Would you be able to share your code with me.  Thanks!
